test/Mono.Linker.Tests/Tests/DocumentationSignatureParserTests.cs

```diff
@@ -0,0 +1,582 @@
+			[ExpectUnresolvedDocumentationSignature ("T:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic<T>")]
+			public class Generic<T>
+			{
+				[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic``1.MethodSyntaxForTypeParameter(`0)")]
+
+				public void MethodSyntaxForTypeParameter (T t)
+				{
+				}
+
+				[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic`1.MethodSyntaxForTypeGenericArgument(Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic{``0})")]
+				public void MethodSyntaxForTypeGenericArgument (Generic<T> g)
+				{
+				}
+
+				[ExpectUnresolvedDocumentationSignature ("P:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic`1.Item(``0)")]
+				public bool this[T t] {
+					get => false;
+					set { }
+				}
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.MethodWithGenericInstantiation(Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic`1)")]
+			public void MethodWithGenericInstantiation (Generic<A> g)
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Method(System.Int32[:,:])")]
+			[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Method(System.Int32[0:,)")]
+			public void Method (int[,] a)
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.NonGenericMethod(``0)")]
+			public void NonGenericMethod (int i)
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("P:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Item(`0)")]
+			[ExpectUnresolvedDocumentationSignature ("P:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Item(``0)")]
+			public int this[int i] {
+				get => 0;
+				set { }
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("M:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.MethodMissingArgumentTypeName(System.)")]
+			public void MethodMissingArgumentTypeName (int i)
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("T:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.")]
+			public class NoType
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("T:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.NoParameterType()")]
+			public void NoParameterType (int i)
+			{
+			}
+
+			[ExpectUnresolvedDocumentationSignature ("T:Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.NoParameterType(Mono.Linker.Tests.DocumentationSignatureParserTests.Invalid.Generic{})")]
+			public void NoGenericParameterType (Generic<A> g)
+			{
+			}
+
+			// these work, but seem like they shouldn't.
+			// see https://github.com/dotnet/roslyn/issues/44315
```

Comment:
Could you please add tracking issue for linker

Reply:
Filed https://github.com/mono/linker/issues/1214